### PR TITLE
ARROW-5056: [Packaging] Adjust conda recipes to use ORC conda-forge package on unix systems

### PR DESCRIPTION
**Pull request by kszucs**

### CODE REVIEW

File: dev/tasks/conda-recipes/pyarrow/meta.yaml [hunk 1]
```diff
@@ -45,14 +45,18 @@ requirements:
...
+    - snappy
     - boost-cpp
     - python
     - setuptools
     - {{ pin_compatible('numpy', lower_bound='1.14') }}
     - pandas
     - six
     - futures  # [py27]
-    - arrow-cpp {{ ARROW_VERSION }}
+    - {{ pin_compatible('arrow-cpp') }}
```
Comment (kszucs) [inline dev/tasks/conda-recipes/pyarrow/meta.yaml]: Pyarrow [misses](https://travis-ci.org/kszucs/crossbow/builds/511950378#L3166) libarrow, and it indeed has not installed in the test environment. I haven't used `run_exports` so far, so I could use a little help here @xhochy :)

File: dev/tasks/conda-recipes/pyarrow/meta.yaml [hunk 2]
```diff
@@ -53,6 +55,9 @@ requirements:
     - six
     - futures  # [py27]
     - arrow-cpp {{ ARROW_VERSION }}
+    # - {{ pin_compatible('arrow-cpp') }}
```
Comment (kszucs) [inline dev/tasks/conda-recipes/pyarrow/meta.yaml]: @xhochy I had to pin arrow-cpp explicitly to pass the builds - I might miss something for pin_compatible to work properly though.

**Comment by kszucs** [comment]
[kszucs/crossbow/build-456](https://github.com/kszucs/crossbow/branches/all?utf8=%E2%9C%93&query=build-546)

File: dev/tasks/conda-recipes/pyarrow/meta.yaml (hunk 2 above)
Comment (xhochy) [reply dev/tasks/conda-recipes/pyarrow/meta.yaml]: I'm also not really sure how it works....

**Review by xhochy — APPROVED** [review]
+1, LGTM

**Comment by kszucs** [comment]
[kszucs/crossbow/build-547](https://github.com/kszucs/crossbow/branches/all?utf8=%E2%9C%93&query=build-547)

**Pull request closed without merging**